packages/integration-sdk-dev-tools/config/husky.js

```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  hooks: {
+    'pre-commit': 'lint-staged',
+    'pre-push': 'yarn test:ci',
```

Comment:
I still long for not-CI test run on `git push`, one that provides reasonable confidence but doesn't make me wait for a CI run on my machine, and then another CI run in the PR, and then another CI run in master. That is, would be great to just run something like `yarn type-check && yarn test`, which I acknowledge may be exactly what `test:ci` is doing ATM. I saw another project somewhere around here that was doing `pre-push: 'yarn prepush'`, that's nice.

Reply:
Okay I have a few ideas on how we can make this nicer. 

For prepush testing, what do you think about using jest's `--changedSince` feature for the `prepush`?
Docs: https://jestjs.io/docs/en/cli.html#--changedsince

I have not personally used it, but it could provide a little bit of confidence while allowing pushes to be quick(er) than a full test run.

Additionally, we could enable typescript's `incremental` option to make typechecking faster. It's on in this repo because we use composite builds, but we can also turn it on projects that do not use `composite` and `references`.

Reply:
That `--changedSince` is pretty interesting, and `incremental` type checking would be awesome where supported as well! Do these kinds of workflow optimizations make sense to you? Others following along? I know there is some value in running the whole shebang.

Reply:
I think the incremental type checking is a clear win. As for `--changedSince`, if it's only used for testing in `prepush` then it should relatively harmless since CI will run the full test suite anyway. I also suspect as developers are making changes to a project they will end up running a full test run out of habit.